src/kthread.cpp

```
@@ -162,6 +167,16 @@ void kt_for(void (*func)(void*, int, int, int), void *data, int n)
 	}
 	for (i = 0; i < t.n_threads; ++i) pthread_join(tid[i], 0);
 
+	if (sp_enabled()) {
+		double busy[MAX_THREADS];
+		int nt = t.n_threads < MAX_THREADS ? t.n_threads : MAX_THREADS;
+		double sum = 0, esum = 0;
+		for (i = 0; i < nt; ++i) { busy[i] = t.w[i].cpu_busy; sum += busy[i]; esum += t.w[i].encode; }
+		g_ktfor.proc_cpu += sum;            /* accumulate across kt_for calls in a step */
+		g_ktfor.encode   += esum;           /* SAM/BAM-build CPU (only worker_sam adds) */
+		sp_thread_stats(&g_ktfor, busy, nt); /* balance stats from the most recent call */
```

Cppcheck (2.21.0)

[warning] Uninitialized variable

(uninitvar)